src/chat/runeMagicCard.ts

```diff
@@ -0,0 +1,360 @@
+          magicPointBoost: flags.formData.magicPointBoost,
+        }),
+      });
+      resultMessages.push({
+        result: ResultEnum.Success,
+        html: getGame().i18n.format("RQG.RuneMagicCard.resultMessageSuccess", {
+          runePointCost: flags.formData.runePointCost,
+          magicPointBoost: flags.formData.magicPointBoost,
+        }),
+      });
+      resultMessages.push({
+        result: ResultEnum.Failure,
+        html: getGame().i18n.format("RQG.RuneMagicCard.resultMessageFailure"),
+      });
+      resultMessages.push({
+        result: ResultEnum.Fumble,
+        html: getGame().i18n.format("RQG.RuneMagicCard.resultMessageFumble", {
+          runePointCost: flags.formData.runePointCost,
+        }),
+      });
+      const result = await Ability.roll(
+        "Cast " + itemData.name,
+        Number(selectedRune.data.chance),
+        Number(flags.formData.ritualOrMeditation) +
+          Number(flags.formData.skillAugmentation) +
+          Number(flags.formData.otherModifiers),
+        speakerName,
+        resultMessages
+      );
+      if (result === ResultEnum.Critical) {
+        // spell takes effect, Rune Points NOT spent, Rune gets xp check, boosting Magic Points spent
+        await RuneMagicCard.SpendRuneAndMagicPoints(
+          0,
+          flags.formData.magicPointBoost,
+          flags.actorId,
+          flags.formData.cultId
+        );
+        await RuneMagicCard.GiveExperienceToRune(actor, flags.formData.selectedRuneId);
+      } else if (result === ResultEnum.Success || result === ResultEnum.Special) {
+        // spell takes effect, Rune Points spent, Rune gets xp check, boosting Magic Points spent
+        await RuneMagicCard.SpendRuneAndMagicPoints(
+          flags.formData.runePointCost,
+          flags.formData.magicPointBoost,
+          flags.actorId,
+          flags.formData.cultId
+        );
+        await RuneMagicCard.GiveExperienceToRune(actor, flags.formData.selectedRuneId);
+      } else if (result === ResultEnum.Failure) {
+        // spell fails, no Rune Point Loss, if Magic Point boosted, lose 1 Magic Point if boosted
+        const boosted = flags.formData.magicPointBoost >= 1 ? 1 : 0;
+        await RuneMagicCard.SpendRuneAndMagicPoints(
+          0,
+          boosted,
+          flags.actorId,
+          flags.formData.cultId
+        );
+      } else if (result === ResultEnum.Fumble) {
+        // spell fails, lose Rune Points, if Magic Point boosted, lose 1 Magic Point if boosted
+        const boosted = flags.formData.magicPointBoost >= 1 ? 1 : 0;
+        await RuneMagicCard.SpendRuneAndMagicPoints(
+          flags.formData.runePointCost,
+          boosted,
+          flags.actorId,
+          flags.formData.cultId
+        );
+      }
+    }
+  }
+
+  private static async GiveExperienceToRune(
+    actor: RqgActor,
+    runeId: string
+  ) {
+    await actor.updateEmbeddedDocuments("Item", [
+      { _id: runeId, data: { hasExperience: true } },
+    ]);
```

Comment:
Getting exp. is nice - we should probably do a ui.notification as well to make the player realise what happened.

Reply:
Created Actor.AwardExperience.  Eventually all experience gain should funnel through this method for consistency.

Reply:
Nice! The only exception would be the POW exp check I think since everything else is items. Maybe add a case for refactoring the other experience checks.